week10/src/pages/Detail.jsx

```diff
@@ -0,0 +1,392 @@
+import heart from "../img/heart.png";
+
+const Detail = () => {
+    const { postId } = useParams();
+    const [details, setDetails] = useState(null);
+    const [isEdit, setIsEdit] = useState(false);
+    const [post, setPost] = useState({ title: "", content: "" });
+    const [heartNum, setHeartNum] = useState(0);
+    const [myHeart, setMyheart] = useState(false);
+    const [myComments, setMyComments] = useState("");
+    const [commentsList, setCommentsList] = useState(null);
+    const [myAccount, setMyAccount] = useState("");
+    const nav = useNavigate();
+
+    useEffect(() => {
+        async function getDetails() {
+            const res = await GetPostDetailApi(postId);
+            setDetails(res);
+            const res2 = await JWTtest();
+
+            setMyAccount(res2.nickname);
+        }
+
+        async function getHeartNumInDetail() {
+            const num = await getHeartNum(postId);
+            setHeartNum(num);
+        }
+
+        async function getCommentsList() {
+            const res = await getComments(postId);
+            setCommentsList(res);
+        }
+
+        async function didIClickHeart() {
+            const HeartHistory = await checkMyClick(postId);
+            setMyheart(HeartHistory);
+        }
+
+        if (!localStorage.getItem("efubtoken")) {
+            alert("로그인 후 이용해주세요.");
+            nav("/register");
+        } else {
+            getDetails();
+            getHeartNumInDetail();
+            getCommentsList();
+            didIClickHeart();
+        }
+    }, [postId]);
+
+    useEffect(() => {
+        async function updateHeartNum() {
+            const num = await getHeartNum(postId);
+            setHeartNum(num);
+        }
+
+        updateHeartNum();
+    }, [myHeart, postId]);
+
+    if (!details) {
+        return <div>로딩 중...</div>;
+    }
+
+    const handleEditBtn = () => {
+        setIsEdit(true);
+    };
+
+    const onSubmitEditPost = async () => {
+        if (post.title.trim() === "") {
+            alert("제목을 입력해주세요.");
+        } else if (post.content.trim() == "") {
+            alert("내용을 입력해주세요.");
+        } else {
+            await EditPostApi(postId, post);
+            setIsEdit(false);
+            window.location.reload();
+        }
+    };
+
+    const onChangesPost = (e) => {
+        const { name, value } = e.target;
+        setPost({ ...post, [name]: value });
+    };
+
+    const deletePost = async () => {
+        await DeletePostApi(postId);
+        alert("게시글 삭제 완료");
+        nav(-1);
+    };
+
+    const onChangesMyComments = (e) => {
+        const { value } = e.target;
+        setMyComments(value);
+    };
+
+    const onSubmitMyComments = async (e) => {
+        if (myComments.trim() === "") {
+            alert("내용을 입력해주세요.");
+        } else {
+            const data = {
+                postId: postId,
+                content: myComments,
+            };
+            await postComments(data);
+            window.location.reload();
```

Comment:
```suggestion
            getCommentsList();
            setMyComments("");
```
해당 코드도 글 수정과 마찬가지로 새로고침 보다는 댓글만 새로 가져와 리렌더링 할 수 있도록 useEffect 외부에서 getCommentsList 함수를 정의하고 여기에서 호출해주세요.
또한 댓글 작성 후에는 댓글 작성 입력을 초기화 해주는 코드를 추가하면 좋을 것 같습니다!